src/v/net/connection.cc

```diff
@@ -133,6 +133,7 @@ connection::connection(
   , _hook(hook)
   , _name(std::move(name))
   , _fd(std::move(f))
+  , _local_addr(_fd.local_address())
```

Comment:
Doing this always is fine I guess as most connections are kafka rpc connections so we will need this sooner than later anyway?

Reply:
The logic is that the local address will never change for the duration of the instance, therefore it is safe to cache